core/org.wso2.carbon.utils/src/main/java/org/wso2/carbon/utils/AuditLog.java

```diff
@@ -18,165 +18,156 @@
-    }
-
-    public String getCorrelationId() {
+    public String getRequestId() {
 
-        return correlationId;
-    }
-
-    public void setCorrelationId(String correlationId) {
-
-        this.correlationId = correlationId;
+        return requestId;
     }
 
     public String getInitiatorId() {
 
         return initiatorId;
     }
 
-    public void setInitiatorId(String initiatorId) {
-
-        this.initiatorId = initiatorId;
-    }
-
-    public String getInitiatorName() {
-
-        return initiatorName;
-    }
-
-    public void setInitiatorName(String initiatorName) {
-
-        this.initiatorName = initiatorName;
-    }
-
     public String getInitiatorType() {
 
         return initiatorType;
     }
 
-    public void setInitiatorType(String initiatorType) {
-
-        this.initiatorType = initiatorType;
-    }
-
-    public String getEventType() {
-
-        return eventType;
-    }
-
-    public void setEventType(String eventType) {
-
-        this.eventType = eventType;
-    }
-
     public String getTargetId() {
 
         return targetId;
     }
 
-    public void setTargetId(String targetId) {
-
-        this.targetId = targetId;
-    }
-
-    public String getTargetName() {
-
-        return targetName;
-    }
-
-    public void setTargetName(String targetName) {
-
-        this.targetName = targetName;
-    }
-
     public String getTargetType() {
 
         return targetType;
     }
 
-    public void setTargetType(String targetType) {
-
-        this.targetType = targetType;
-    }
-
-    public String getDataChange() {
-
-        return dataChange;
-    }
-
-    public void setDataChange(String dataChange) {
-
-        this.dataChange = dataChange;
+    public String getAction() {
+
+        return action;
+    }
+
+    public Map<String, Object> getData() {
+
+        return data;
+    }
+
+    /**
+     * Builder class for AuditLog.
+     * This class follows the Builder design pattern, providing a way to construct a AuditLog object step by step.
+     */
+    public static class AuditLogBuilder {
+
+        private String id;
+        private String recordedAt;
+        private String requestId;
+        private final String initiatorId;
+        private final String targetId;
+        private final String initiatorType;
+        private final String targetType;
+        private final String action;
+        private Map<String, Object> data;
+
+        public AuditLogBuilder(String initiatorId, String initiatorType, String targetId, String targetType, String action) {
+
+            this.initiatorId = initiatorId;
+            this.targetId = targetId;
+            this.initiatorType = initiatorType;
+            this.targetType = targetType;
+            this.action = action;
+        }
+
+        public AuditLog.AuditLogBuilder putData(String key, Object value) {
+
+            if (this.data == null) {
+                this.data = new HashMap<>();
+            }
+
+            if (value != null) {
```

Comment:
this could be checked before map's null check (in the form of a guard clause) to avoid initializing map in case of single null valued data item

Reply:
fixed it